backend/internal/controllers/book.go

```diff
@@ -46,3 +48,123 @@ func (c *BookController) GetBookDetails(rw http.ResponseWriter, req *http.Reques
 
 	writeResponse(rw, http.StatusOK, bookDetails)
 }
+
+// Creates a book and inserts it into the book database (/books)
+func (c *BookController) CreateBook(rw http.ResponseWriter, req *http.Request) {
+	var reqBook models.APICreateBook
+	var resBook models.Book
+	err := json.NewDecoder(req.Body).Decode(&reqBook)
+	if err != nil {
+		writeResponse(rw, http.StatusBadRequest, "error")
```

Comment:
Can you replace "error" with a "invalid request schema" message or similar? Same below.